addons/l10n_id/models/res_bank.py

```diff
@@ -0,0 +1,67 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import requests
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+QRIS_API_URL = "https://qris.online/restapi/qris/show_qris.php"
+
+class ResBank(models.Model):
+    _inherit = "res.partner.bank"
+
+    l10n_id_qris_api_key = fields.Char("QRIS API Key")
+    l10n_id_qris_mid = fields.Char("QRIS Merchant ID")
+
+    @api.model
+    def _get_available_qr_methods(self):
+        rslt = super()._get_available_qr_methods()
+        rslt.append(('id_qr', _("QRIS"), 40))
+        return rslt
+
+    def _get_error_messages_for_qr(self, qr_method, debtor_partner, currency):
+        if qr_method == 'id_qr' and self.country_code == 'ID':
+            if currency.name not in ['IDR']:
+                return _("Can't generate a QR code with a currency other than IDR.")
+            if not (self.l10n_id_qris_api_key and self.l10n_id_qris_mid):
+                return _("Please configure the QRIS API Key and Merchant ID on the bank's configuration")
```

Comment:
```suggestion
                return _("To use QRIS QR code, Please configure the QRIS API Key and Merchant ID on the bank's configuration")
```
Or something similar